app/lib/screens/app_onboarding/loading_screen.dart

```diff
@@ -105,7 +114,37 @@ class _LoadingScreenState extends State<LoadingScreen> {
           StudyULogger.fatal('Could not login and retrieve the study subject. '
               'One reason for this might be that the study subject is no '
               'longer available and only resides in app backup');
-          throw Exception("Remote subject not found");
+          // throw Exception("Remote subject not found");
+          // Ask the user if they want to delete all secure storage data or try to re-read
+          // show popup dialog to the user
+          if (!mounted) return null;
+          final result = await showDialog<bool>(
+            context: context,
+            builder: (context) {
+              return AlertDialog(
+                title: const Text("Secure Storage Error"),
+                content: const Text(
+                  "The secure storage seems to be corrupted. Do you want to delete all secure storage data?",
+                ),
+                actions: [
+                  TextButton(
+                    onPressed: () => Navigator.of(context).pop(false),
+                    child: const Text("Cancel"),
+                  ),
+                  TextButton(
+                    onPressed: () => Navigator.of(context).pop(true),
+                    child: const Text("Delete All Data"),
+                  ),
+                ],
+              );
+            },
+          );
+          if (result == true) {
+            await SecureStorage.deleteAll();
+            StudyULogger.info("Secure storage data deleted");
+          } else {
+            StudyULogger.info("User chose not to delete secure storage data");
+          }
```

Comment:
Initially, when re-installing the app, due to a bug in flutter secure storage the storage might have been unreadable, due to a BadPaddingException. Other causes such as switching the database with development versions or other unknown reasons might also lead to this point, where an Exception is thrown but the user is not getting redirected and is stuck in the loading screen. With the new prompt, the user may reset the secure storage and start over.